flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/hints/stream/StateTtlHintTest.java

```diff
@@ -0,0 +1,207 @@
+                "select /*+ STATE_TTL('T2' = '2d', 'T1' = '1d') */* from T1 join T2 on T1.a1 = T2.a2";
+        verify(sql);
+    }
+
+    @Test
+    void testJoinStateTtlHintOnlyWithRightSide() {
+        String sql = "select /*+ STATE_TTL('T2' = '2d') */* from T1 join T2 on T1.a1 = T2.a2";
+        verify(sql);
+    }
+
+    @Test
+    void testJoinStateTtlHintWithContinuousJoin() {
+        String sql =
+                "select /*+ STATE_TTL('T2' = '2d', 'T3' = '3d', 'T1' = '1d') */* from T1, T2, T3 where T1.a1 = T2.a2 and T2.b2 = T3.b3";
+        verify(sql);
+    }
+
+    @Test
+    void testJoinStateTtlHintWithMultiLevelJoin() {
+        String sql =
+                "select /*+ STATE_TTL('T2' = '2d', 'T3' = '3d', 'T1' = '1d') */* from T1 "
+                        + "join (select T2.* from T2 join T3 on T2.b2 = T3.b3) TMP on T1.a1 = TMP.b2";
+        assertThatThrownBy(() -> verify(sql))
+                .isInstanceOf(ValidationException.class)
+                .hasMessageContaining(
+                        "The options of following hints cannot match the name of input tables or views: \n`%s` in `%s`",
+                        "T2, T3", "STATE_TTL");
+    }
+
+    @Test
+    void testJoinStateTtlHintWithOneUnknownTable() {
+        String sql =
+                "select /*+ STATE_TTL('T5' = '2d', 'T1' = '1d') */* from T1 join T2 on T1.a1 = T2.a2";
+
+        assertThatThrownBy(() -> verify(sql))
+                .isInstanceOf(ValidationException.class)
+                .hasMessageContaining(
+                        "The options of following hints cannot match the name of input tables or views: \n`%s` in `%s`",
+                        "T5", "STATE_TTL");
+    }
+
+    @Test
+    void testJoinStateTtlHintWithTwoUnknownTables() {
+        String sql =
+                "select /*+ STATE_TTL('T5' = '2d', 'T6' = '1d') */* from T1 join T2 on T1.a1 = T2.a2";
+
+        assertThatThrownBy(() -> verify(sql))
+                .isInstanceOf(ValidationException.class)
+                .hasMessageContaining(
+                        "The options of following hints cannot match the name of input tables or views: \n`%s` in `%s`",
+                        "T5, T6", "STATE_TTL");
+    }
+
+    @Test
+    void testJoinStateTtlHintWithView() {
+        String sql =
+                "select /*+ STATE_TTL('T1' = '2d', 'V4' = '1d') */* from T1 join V4 on T1.a1 = V4.a4";
+        verify(sql);
+    }
+
+    @Test
+    void testJoinStateTtlHintWithUnknownView() {
+        String sql =
+                "select /*+ STATE_TTL('T1' = '2d', 'V8' = '1d') */* from T1 join V4 on T1.a1 = V4.a4";
+        assertThatThrownBy(() -> verify(sql))
+                .isInstanceOf(ValidationException.class)
+                .hasMessageContaining(
+                        "The options of following hints cannot match the name of input tables or views: \n`%s` in `%s`",
+                        "V8", "STATE_TTL");
+    }
+
+    @Test
+    void testMultiJoinStateTtlHint() {
```

Comment:
`testMultiJoinStateTtlHint` -> `testDuplicateJoinStateTtlHint`?

Is it a better idea to throw exceptions here?  
I think the `STATE_TTL` hint is different from the join hint on this point.
Duplicate state ttl hint with contradictable hint values may result in
unexpected results, while for join strategy, it's more about the performance.
WDYT?
```
stateTtlHints=[[[STATE_TTL options:{RIGHT=2d}][STATE_TTL options:{LEFT=1d, RIGHT=8d}]]]
```

Reply:
I agree there will be some correctness issues if the user uses the state ttl hint but the actual operator state ttl is different from expected. But query hint originally has the semantics of "suggestion", isn't it?
I think having two different behaviors for join hints and state TTL hints, which are both types of query hints, could create some misunderstanding among users.
If we need to explicitly throw an exception in a scenario where state ttl hint conflicts, then we also need to explicitly throw an exception in a scenario where state ttl hint is used but does not actually take effect. Throwing exceptions so frequently will harm users.
Perhaps what we really need is how to correctly convey the warning information that this query hint is not used correctly to the user.

Reply:
Take `testJoinStateTtlHintWithDuplicatedArgs` for example. If we really want to align the behavior with other join hints like `BROACAST`, we need to make sure always apply the first provided hint value. However, from the explained result, the 1d appears first. Besides, there's no end-to-end test to ensure this.

Reply:
Thanks for pointing out this obscure issue.
Regarding the issue where kv hints do not choose the first kv in the event of a conflict while list hints choose the first hint, this will be uniformly fixed by FLINK-34034.
In this PR, I have modified the code so that when fetching state ttl hints with the same key, it is now based on the first kv.